corehq/apps/accounting/static/accounting/js/stripe.js

```diff
@@ -4,11 +4,11 @@ self.stripePromise = undefined;
 self.cardElement = undefined;
 
 function getCardElementPromise(key) {
-    if (!key) {
-        throw new Error("Cannot load Stripe, key not provided");
-    }
-
     let promise = new Promise((resolve) => {
+        if (!key) {
+            console.warn("Cannot load Stripe, key not provided");
+            return;
+        }
```

Comment:
How will this affect [code that operates on the returned value](https://github.com/dimagi/commcare-hq/blob/7a821d1323fc4ad8f8d81755a2320d9fc1631f45/corehq/apps/accounting/static/accounting/js/stripe_card_manager.js#L17-L21)? It looks like this code will raise a more cryptic error with this change, because the `then` clause will execute with `null`, right?

I'm curious if this means we should be updating the callers as well, or if we should be changing this to reject the promise, rather than resolve.

Reply:
Right now the `then`s in calling code will never execute, because the promise never gets resolved. I can see an argument for rejecting the promise if the key isn't there, but it also means adding code to all of the callers because right now they don't handle rejections. I kind of like keeping the code simpler.

Reply:
Thanks, I just ran Promise code locally and noticed the same thing. I assumed that a promise like `Promise(function(resolve) { return })` would return `undefined` to code like `promise.then(result => ...)`, but realized that its like you mentioned: the Promise is left in a "pending" state. 

I can see that making conceptual sense. All environments try to fetch the stripe environment, and local ones just never succeed. It's probably cleaner if local environments never try, but this is a working compromise for now